### PR TITLE
Refactor mission log scrollback

### PR DESCRIPTION
**Pull request by MjnMixael**
Reorganizes how the mission log scrollback is initialized and drawn. Instead of separate linked lists for timestamp and entries, this uses a single vector for each line. The vector entry holds all the required information required for drawing, including text, color, etc. This will make SCPUI access much easier in a follow-up PR.

Tested viewing the scrollback with a small log (no scrolling required), a large log, and with as many different entry types as I could reasonably come up with. Each type of log entry seems to work exactly as it did before.

### CODE REVIEW

**Review by wookieejedi — APPROVED** [review]
My testing shows this all works well, and overall looks good. Also provides some useful utility in that once it is merged, it will make issues such as #5187 easier to address. Also noting that it is always useful to have another set of eyes on the code, too.

**Review by The-E — APPROVED** [review]

**Pull request merged**
